Userland/Libraries/LibCompress/Xz.cpp

```diff
@@ -217,7 +217,7 @@ ErrorOr<bool> XzDecompressor::load_next_stream()
             // Read the first byte until we either get a non-null byte or reach EOF.
             auto byte_or_error = m_stream->read_value<u8>();
 
-            if (byte_or_error.is_error() && byte_or_error.error().is_custom_error() && byte_or_error.error().custom_error() == Error::CustomError::NotEnoughData)
```

Comment:
If this is the reason why we need operator==, have you considered making CustomError an enum class?

Reply:
Does `enum` vs. `enum class` make a difference here? The reason why I added the equality operator is only to skip having to check `is_custom_error` first, and I added one for POSIX errors purely to make it equal to the new custom error type.